pkg/option/resolver/resolver.go

```diff
@@ -0,0 +1,285 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+// Package resolver provides the logic for merging in the various sources of configuration,
+// overrides, and drop-ins.
+package resolver
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+	"time"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	labels "k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/sets"
+	apivalidation "k8s.io/apimachinery/pkg/util/validation"
+
+	ciliumv2alpha1 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2alpha1"
+	"github.com/cilium/cilium/pkg/k8s/client"
+	"github.com/cilium/cilium/pkg/logging"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+)
+
+const (
+	KindConfigMap  = "config-map"
+	KindNode       = "node"
+	KindNodeConfig = "cilium-node-config"
+)
+
+type ConfigSource struct {
+	Kind      string // one of config-map, overrides, node
+	Namespace string // The namespace for the ConfigMap or CiliumNodeConfigs
+	Name      string // The name of the ConfigMap or Node, unused for Overrides
+}
+
+var log = logging.DefaultLogger.WithField(logfields.LogSubsys, "config-resolver")
```

Comment:
nit: Just FYI, as a convention we typically define the subsystem derived from the directory path, so in this case `option-resolver`.

Reply:
fixed.